### ISSUE TITLE
Calibre not moving books to book folder

### ISSUE THREAD

**Issue by LordJay23** [DobyTang/LazyLibrarian#1176]
So since a recent update, LL stopped sending books to the destination folder using calibredb.  

Here's a snippet:

`08-Jan-2018 04:13:22 - DEBUG   :: POSTPROCESS : postprocess.py:processDir:438 : Found folder (100%) [/mnt/downloads
/NewSync/lazylibrarian/The Call of Cthulhu and Other Weird Stories - H P Lovecraft (EPUB).epub] for eBook The Call 
of Cthulhu and Other Weird Stories - H P Lovecraft (EPUB).epub
08-Jan-2018 04:13:22 - DEBUG   :: POSTPROCESS : postprocess.py:processDir:491 : Found match (100%): /mnt/downloads/
NewSync/lazylibrarian/The Call of Cthulhu and Other Weird Stories - H P Lovecraft (EPUB).epub for eBook The Call of
 Cthulhu and Other Weird Stories - H P Lovecraft (EPUB).epub
08-Jan-2018 04:13:23 - DEBUG   :: POSTPROCESS : postprocess.py:processDir:497 : Processing eBook 35372701
08-Jan-2018 04:13:24 - DEBUG   :: POSTPROCESS : postprocess.py:processDestination:1082 : Importing The Stories of H
. P. Lovecraft - H.P. Lovecraft into calibre library
08-Jan-2018 04:13:24 - DEBUG   :: POSTPROCESS : calibre.py:calibredb:420 : [u'/usr/local/bin/calibredb', 'add', '-1
', '--with-library', u'/mnt/collection/Books/Organized/Main', u'/mnt/downloads/NewSync/lazylibrarian/The Call of Ct
hulhu and Other Weird Stories - H P Lovecraft (EPUB).epub']
08-Jan-2018 04:14:08 - DEBUG   :: POSTPROCESS : postprocess.py:processDestination:1107 : /usr/local/bin/calibredb r
esult : Added book ids: 2880
08-Jan-2018 04:14:08 - DEBUG   :: POSTPROCESS : postprocess.py:processDestination:1108 : /usr/local/bin/calibredb e
rror  : /mnt/collection/Books/Organized/Main
08-Jan-2018 04:14:08 - DEBUG   :: POSTPROCESS : postprocess.py:processDestination:1118 : Calibre ID: 2880
08-Jan-2018 04:14:09 - DEBUG   :: POSTPROCESS : postprocess.py:processOPF:1490 : Saved metadata to: /mnt/downloads/
NewSync/lazylibrarian/The Call of Cthulhu and Other Weird Stories - H P Lovecraft (EPUB).epub/The Stories of H. P. Lovecraft - H.P. Lovecraft.opf
08-Jan-2018 04:14:09 - DEBUG   :: POSTPROCESS : calibre.py:calibredb:420 : [u'/usr/local/bin/calibredb', 'set_metadata', '--with-library', u'/mnt/collection/Books/Organized/Main', '2880', u'/mnt/downloads/NewSync/lazylibrarian/The Call of Cthulhu and Other Weird Stories - H P Lovecraft (EPUB).epub/The Stories of H. P. Lovecraft - H.P. Lovecraft.opf']
08-Jan-2018 04:14:36 - DEBUG   :: POSTPROCESS : postprocess.py:processDestination:1135 : /usr/local/bin/calibredb set opf reports: Title               : The Stories of H. P. Lovecraft
08-Jan-2018 04:14:36 - DEBUG   :: POSTPROCESS : postprocess.py:processDestination:1160 : Calibre trying directory [/mnt/collection/Books/Organized/Main/H.P. Lovecraft/The Stories of H. P. Lovecraft (2880)]
08-Jan-2018 04:14:36 - INFO    :: POSTPROCESS : librarysync.py:LibraryScan:320 : Scanning eBook directory: /mnt/collection/Books/Organized/Main/H.P. Lovecraft/The Stories of H. P. Lovecraft (2880)
08-Jan-2018 04:14:44 - INFO    :: POSTPROCESS : librarysync.py:LibraryScan:337 : Missing eBooks will be marked as Skipped
08-Jan-2018 04:14:44 - DEBUG   :: POSTPROCESS : librarysync.py:LibraryScan:443 : [/mnt/collection/Books/Organized/Main/H.P. Lovecraft/The Stories of H. P. Lovecraft (2880)] Now scanning subdirectory 
08-Jan-2018 04:14:44 - DEBUG   :: POSTPROCESS : librarysync.py:LibraryScan:477 : book meta [9781101097144] [en] [S. T. Joshi, H. P. Lovecraft] [The Call of Cthulhu and Other Weird Stories] [epub]
08-Jan-2018 04:14:44 - DEBUG   :: POSTPROCESS : librarysync.py:LibraryScan:564 : Found Language [en] ISBN [9781101097144]
08-Jan-2018 04:14:44 - DEBUG   :: POSTPROCESS : librarysync.py:LibraryScan:576 : Already cached Lang [en] ISBN [110]
08-Jan-2018 04:14:44 - DEBUG   :: POSTPROCESS : formatter.py:formatAuthorName:366 : Formatted authorname [S. T. Joshi, H. P. Lovecraft] to [H. P. Lovecraft S. T. Joshi]
08-Jan-2018 04:14:44 - DEBUG   :: POSTPROCESS : formatter.py:formatAuthorName:376 : Stripped authorname [H. P. Lovecraft S. T. Joshi] to [H.P. Lovecraft S. T. Joshi]
08-Jan-2018 04:14:45 - DEBUG   :: POSTPROCESS : importer.py:addAuthorNameToDB:49 : Author H.P. Lovecraft S. T. Joshi not found in database, trying to add
08-Jan-2018 04:14:45 - DEBUG   :: POSTPROCESS : gr.py:find_author_id:242 : Searching for author with name: H.P. Lovecraft S. T. Joshi
08-Jan-2018 04:14:45 - DEBUG   :: POSTPROCESS : cache.py:get_cached_request:160 : CacheHandler: Storing XML for https://www.goodreads.com/api/author_url/H.P.%20Lovecraft%20S.%20T.%20Joshi?key=ckvsiSDsuqh7omh74ZZ6Q
08-Jan-2018 04:14:45 - DEBUG   :: POSTPROCESS : cache.py:get_cached_request:137 : CacheHandler: Returning CACHED response for https://www.goodreads.com/author/show/9494.xml?key=ckvsiSDsuqh7omh74ZZ6Q
`
It's creating the book folder, and moving the .opf file and any images, but the book itself is not going.  I feel like I'm missing something.

This is with yesterday's build of philborman, and calibredb 3.5

**Comment by LordJay23** [comment]
Ok, I see what calibre.py is returning, and the error isn't really an error, it's just showing the dest_url if the command succeeded.

**Comment by philborman** [comment]
Yes, we should probably suppress that line if it's empty.
I think the problem might be that the author isn't an author, it's two...  [H. P. Lovecraft S. T. Joshi]
Not sure how calibre handles this. It imports the book under it's internal id 2880. We send it the opf and the image, but maybe it files the book under Joshi rather than Lovecraft?  If that is what's happening  what would be the best way to handle it, do you think? Multiple authors is tricky to do properly...

**Comment by LordJay23** [comment]
I've tried though, it's doing that on every download that's come through in the last month, including single author titles.

**Comment by philborman** [comment]
Hmm, strange. Calibre says it imported the book, and tells us the book id it assigned to it, so it should be there somewhere.
It's a bit confusing. From your log above it seems calibre **did** import the book and placed it in the folder [/mnt/collection/Books/Organized/Main/H.P. Lovecraft/The Stories of H. P. Lovecraft (2880)] 
Lazylibrarian scanned that folder and found an epub file there 
book meta [9781101097144] [en] [S. T. Joshi, H. P. Lovecraft] [The Call of Cthulhu and Other Weird Stories] [epub]
but as I said earlier, in this particular book there are two authors listed which could be the issue in this case.  You don't seem to get a full reply from calibre set opf, it only seems to report the title back, nothing else.

Could you try ticking the "only add ebook, not opf or jpg" button in lazylibrarian config->processing. If that is ticked we send separate commands to force the author and title to be what we want. Should get more debug logging that way which might point us in the right direction.

**Comment by LordJay23** [comment]
Will do.

**Comment by LordJay23** [comment]
It didn't actually create anything, but it did add an entry to the database.  Once I tried to browse to it inside calibre, it created the folder and a metadata.opf file.  But still no actual file. 

So, that library path does include a symlink, but I'm able to run the calibredb command from the command line and get the expected result.  However, just to eliminate the possibility, I'm going to take the symlink out of the equation, and try again.

**Comment by LordJay23** [comment]
Nothing.  Removed all symlinks.  It says it's creating the book, and it does in the database.  It's just not creating the book folder until I try to access it in calibre, and then it does.  LL thinks it's successfully processing the book.   Just no file...

Last ditch... I spotted a line that it was copying the file to my calibre auto-import folder.  I looked, and all my book files are sitting there.  I'm not USING calibre auto import.  Cleared that out, tried it again, and suddenly they're going where they're supposed to go :)

Might want to note that it's an either/or :)
